mne/source_estimate.py

```diff
@@ -1931,6 +1931,8 @@ def _morph_buffer(data, idx_use, e, smooth, n_vertices, nearest, maps,
               % (len(data_sum) - len(idx_use), len(data_sum)))
 
     logger.info('    %d smooth iterations done.' % (k + 1))
+    if maps is None:
+        return data
```

Comment:
If we're going to do this, then we should also cause it to use this short-circuit in `stc.morph(...)` whenever `subject_to` and `subject_from` are the same. Alternatively, you can change your use of the code here to use the proper maps (which should just be sparse identity matrices). But they should both use the same code path in any case.